tools/text_processing/text_processing/awk.xml

```diff
@@ -1,34 +1,61 @@
-<tool id="tp_awk_tool" name="Text reformatting" version="@BASE_VERSION@.1">
+<tool id="tp_awk_tool" name="Text reformatting" version="@BASE_VERSION@.2">
     <description>with awk</description>
     <macros>
         <import>macros.xml</import>
     </macros>
     <requirements>
         <requirement type="package" version="4.2.0">gawk</requirement>
+        <requirement type="package" version="8.30">coreutils</requirement>
     </requirements>
     <version_command>awk --version | head -n 1</version_command>
     <command>
 <![CDATA[
+        #set $tool_input = 1
+        env -i -S "$envs"
         awk
             --sandbox
             -v FS='	'
             -v OFS='	'
             --re-interval
             -f "$awk_script"
-            "$infile"
+            #if $infile and not $last
+                tool_input=0
+                "$infile"
+            #end if
+            #for group in $infiles
+                tool_input=$tool_input
+                #set $tool_input+=1
+                #for file in $group.infile
+                    "$file"
+                #end for
+            #end for
+            #if $infile and $last
+                tool_input=0
+                "$infile"
+            #end if
         > "$outfile"
 ]]>
     </command>
     <configfiles>
         <configfile name="awk_script">$code</configfile>
     </configfiles>
     <inputs>
-        <param name="infile" format="txt" type="data" label="File to process" />
+        <param name="infile" format="txt" type="data" optional="true" label="Single file to process" help="A seperate execution will occur for each provided file"/>
+        <param name="last" type="boolean" checked="false" label="Include the above file last" help="Include the above input after any of the below inputs" />
+        <repeat name="infiles" title="Multiple input files">
+            <param name="infile" format="txt" type="data" multiple="true" label="File to process" />
+        </repeat>
         <param name="code" type="text" area="true" size="5x35" label="AWK Program" help="">
+            <sanitizer sanitize="false" />
```

Comment:
I think that changing this `sanitizer` may introduce security issues, can you restore the previous version?

Reply:
Can you explain? I need to disable the ${} substitution

Reply:
I wrongly thought `$code` was ending up in `<command/>`, in `<configfile/>` should be fine.